### DISCUSSION TITLE
about the websocket in libdatachannal

### DISCUSSION THREAD

**Discussion by bestxpp** [paullouisageneau/libdatachannel#1088] [Q&A] [answered]
The WebSocket send interface in libdatachannal, does it copy the data? Is it sent asynchronously? Is it thread-safe?

**Comment by paullouisageneau** [comment, marked as answer]
The method `send(message_variant data)` does not copy binary data (the library uses move semantics where possible), but `send(const byte *data, size_t size)` does copy the data. The library will attempt to send synchronously: if the data can be sent synchronously, `send()` returns true, but if it's not possible because the underlying socket buffer is full (meaning you are sending too fast), `send()` returns false, and the data is kept buffered and will be sent it asynchronously when possible. In any case, `send()` does not block on network. The method is also thread-safe, meaning you can send concurrently from different threads.

**Reply by bestxpp** [reply]
![image](https://github.com/paullouisageneau/libdatachannel/assets/28625121/844bf05d-fcde-4118-9fe2-f251acf93e83)


![image](https://github.com/paullouisageneau/libdatachannel/assets/28625121/f5e6d3f2-dff4-4e50-9e67-e9cf4c47d419)
 

When the data I send exceeds 65535, it throws an exception

**Reply by paullouisageneau** [reply]
The max message size is 65535 bytes by default for WebSockets. Since https://github.com/paullouisageneau/libdatachannel/pull/1086 on `master`, you can change it with `maxMessageSize` in `WebSocket::Configuration`.

**Comment by bestxpp** [comment]
> 默认情况下，WebSocket 的最大消息大小为 65535 字节。从#1086开始，您可以使用in`master`更改它。`maxMessageSize``WebSocket::Configuration`

thx, the framework now supports custom message sizes
